udp_mux_multi_test.go

```diff
@@ -110,46 +111,107 @@ func testMultiUDPMuxConnections(t *testing.T, udpMuxMulti *MultiUDPMuxDefault, u
 	}
 }
 
+type muxCase struct {
+	ports []int
+	opts  []UDPMuxFromPortOption
+}
+
 func TestUnspecifiedUDPMux(t *testing.T) {
 	report := test.CheckRoutines(t)
 	defer report()
 
 	lim := test.TimeOut(time.Second * 30)
 	defer lim.Stop()
 
-	muxPort := 7778
-	udpMuxMulti, err := NewMultiUDPMuxFromPort(muxPort, UDPMuxFromPortWithInterfaceFilter(func(s string) bool {
+	cases := map[string]muxCase{
+		"single port": {ports: []int{7778}},
+		"multi ports": {ports: []int{7779, 7780, 7781}},
+		"batch write": {ports: []int{7782}, opts: []UDPMuxFromPortOption{UDPMuxFromPortWithBatchWrite(10, 2*time.Millisecond)}},
+	}
+
+	for name, val := range cases {
+		cname, muxCase := name, val
+		t.Run(cname, func(t *testing.T) {
+			opts := []UDPMuxFromPortOption{
+				UDPMuxFromPortWithInterfaceFilter(func(s string) bool {
+					defaultDockerBridgeNetwork := strings.Contains(s, "docker")
+					customDockerBridgeNetwork := strings.Contains(s, "br-")
+					return !defaultDockerBridgeNetwork && !customDockerBridgeNetwork
+				}),
+				UDPMuxFromPortWithReadBufferSize(1024 * 1024),
+				UDPMuxFromPortWithWriteBufferSize(1024 * 1024),
+			}
+			opts = append(opts, muxCase.opts...)
+			udpMuxMulti, err := NewMultiUDPMuxFromPorts(muxCase.ports, opts...)
+			require.NoError(t, err)
+
+			require.GreaterOrEqual(t, len(udpMuxMulti.muxes), 1, "at least have 1 muxes")
+			defer func() {
+				_ = udpMuxMulti.Close()
+			}()
+
+			wg := sync.WaitGroup{}
+
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				testMultiUDPMuxConnections(t, udpMuxMulti, "ufrag1", udp)
+			}()
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				testMultiUDPMuxConnections(t, udpMuxMulti, "ufrag2", udp4)
+			}()
+
+			// Skip IPv6 test on i386
+			const ptrSize = 32 << (^uintptr(0) >> 63)
```

Comment:
Maybe we should check against `runtime.GOARCH` here?

Reply:
I think they are identical, and if one day there is a new GOARCH, the test will be failed.